recipes/linalg-complex/24.11/mc/repo/packages/trilinos/package.py

```diff
@@ -582,13 +582,6 @@ def url_for_version(self, version):
         url = "https://github.com/trilinos/Trilinos/archive/refs/tags/trilinos-release-{0}.tar.gz"
         return url.format(version.dashed)
 
-    def setup_dependent_run_environment(self, env, dependent_spec):
-        if "+cuda" in self.spec:
-            # currently Trilinos doesn't perform the memory fence so
-            # it relies on blocking CUDA kernel launch. This is needed
-            # in case the dependent app also run a CUDA backend via Trilinos
-            env.set("CUDA_LAUNCH_BLOCKING", "1")
-
```

Comment:
To do: This will be removed/modified in the upstream spack package in the near future. I will update the bundled trilinos definition to match that before this PR is ready for merging.

Reply:
This was fixed upstream in https://github.com/spack/spack/pull/49468. I've updated the trilinos package definitions to match the upstream in https://github.com/eth-cscs/alps-uenv/pull/188/commits/9debcf54511eb6489dfde1a469d400f97c70f889.